### PR TITLE
Coordinates field (task #14621)

### PR DESCRIPTION
**Pull request by eliamariutti**
image
image

New coordinates filed with GPS pickup helper in modal, map in view and custom validation.

### CODE REVIEW

File: src/FieldHandlers/Provider/ValidationRules/CoordinatesValidationRules.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class CoordinatesValidationRules extends AbstractValidationRules
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function provide($validator = null, array $options = [])
+    {
+        $validator = parent::provide($validator, $options);
+
+        $validator->add($options['fieldDefinitions']->getName(), 'validRegex', [
```
Comment (georgeconstantinou) [inline src/FieldHandlers/Provider/ValidationRules/CoordinatesValidationRules.php]: @eliamariutti lets replace this with CakePHP's validation rules: `$validator->longitude()` and `$validator->latitude()`.

Comment (eliamariutti) [reply src/FieldHandlers/Provider/ValidationRules/CoordinatesValidationRules.php]: @georgeconstantinou I add more strict rules on the validation. Cakephp offer latLong(), geoCoordinate() etc... but all those don't prevent to have a space in the middle of the two values, That's why I added a custom role.

File: src/Template/Element/FieldHandlers/CoordinatesFieldHandler/input.ctp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+use Cake\Core\Configure;
+
+$api_key = Configure::read("CsvMigrations.coordinates_field.GoogleMapKey");
+
+if (empty($api_key)) {
+    $attributes += [
+        'type' => $type,
+        'label' => $label,
```
Comment (softius) [inline src/Template/Element/FieldHandlers/CoordinatesFieldHandler/input.ctp]: @eliamariutti this needs to be wrapped with `_d('Qobo/CsvMigrations', ...)`

File: src/Template/Element/FieldHandlers/CoordinatesFieldHandler/input.ctp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        'block' => 'scriptBottom'
+    ]
+);
+
+$attributes = isset($attributes) ? $attributes : [];
+
+$randId = rand(1, 1000000);
+$attributes += [
+    'type' => $type,
+    'label' => $label,
```
Comment (softius) [inline src/Template/Element/FieldHandlers/CoordinatesFieldHandler/input.ctp]: @eliamariutti this needs to be wrapped with `_d('Qobo/CsvMigrations', ...)`

File: src/Template/Element/FieldHandlers/CoordinatesFieldHandler/input.ctp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+echo $this->Form->control($name, $attributes);
+?>
+
+<div class="modal fade coordinates_modal" id="modal-default-<?php echo $randId; ?>">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span></button>
+        <h4 class="modal-title">GPS : <span class="modal_gps_value"><?php echo $value; ?></span></h4>
```
Comment (softius) [inline src/Template/Element/FieldHandlers/CoordinatesFieldHandler/input.ctp]: @eliamariutti let's rename GPS to Coordinates and wrap with `_d('Qobo/CsvMigrations', ...)`

File: src/Template/Element/FieldHandlers/CurrencyFieldHandler/input.ctp [hunk 5]
```diff
@@ -10,23 +10,6 @@
  * @license       https://opensource.org/licenses/mit-license.php MIT License
  */
 
-$this->Html->script(
```
Comment (softius) [inline src/Template/Element/FieldHandlers/CurrencyFieldHandler/input.ctp]: @eliamariutti why do we edit the CurrentyFieldHandler here?

Comment (eliamariutti) [reply src/Template/Element/FieldHandlers/CurrencyFieldHandler/input.ctp]: @softius it is just a clean up, we don't need to include those script here.

**Pull request merged**